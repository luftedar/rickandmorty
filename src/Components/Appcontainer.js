import { Route, Switch } from 'react-router-dom';
import Characterdetails from './Characterdetails';
import Characters from './Characters';
import Home from './Home';

function Appcontainer() {
  return (
    <div className="app-container">
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/characters">
          <Characters />
        </Route>
        <Route path="/character/:id">
          <Characterdetails />
        </Route>
      </Switch>
    </div>
  );
}

export default Appcontainer;
