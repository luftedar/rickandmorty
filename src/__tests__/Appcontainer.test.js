import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { Provider } from 'react-redux';
import store from '../__mocks__/configureStore';
import Appcontainer from '../Components/Appcontainer';

describe('App page is working fine', () => {
  test('renders App', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Appcontainer />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
