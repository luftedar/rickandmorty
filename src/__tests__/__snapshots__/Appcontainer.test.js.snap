// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page is working fine renders App 1`] = `
<div
  className="app-container"
>
  <div
    className="home-page"
  >
    <a
      className="home-link d-flex d-center"
      href="/characters"
      onClick={[Function]}
    >
      Characters
    </a>
  </div>
</div>
`;
